### PR TITLE
[Merged by Bors] - Some Improvements to intelliJ extension

### PR DESCRIPTION
**Pull request by jamowei**
* same default options as CLI
* improved UI layouts
* fixed not reappearing window after pressing cancel-button
* separate tcp and udp outgoing option
* allow only single selection for namespaces and pods
* refactor UI-creating code to a more Kotlin functional style

### CODE REVIEW

**Comment by infiniteregrets** [comment]
hi @jamowei, thanks again for improving the extension!

Just a quick note on the namespaces,

They look like this for me:
Screen Shot 2022-10-26 at 11 37 05 AM

even though I have a few more namespaces

File: intellij-ext/src/main/kotlin/com/metalbear/mirrord/LogLevel.kt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package com.metalbear.mirrord
+
+enum class LogLevel {
+    ERROR, WARN, INFO, DEBUG, TRACE;
+}
```
Comment (infiniteregrets) [inline intellij-ext/src/main/kotlin/com/metalbear/mirrord/LogLevel.kt:5]: The reason we kept this as a text field was to allow the user to choose whatever logs they wanna see, forexample, 
RUST_LOG=mirrord=debug, warn

Comment (infiniteregrets) [reply intellij-ext/src/main/kotlin/com/metalbear/mirrord/LogLevel.kt:5]: @eyalb181 wdyt? should we keep this as a dropdown or a text field

**Review by infiniteregrets — COMMENTED** [review]
Looks great!

File: intellij-ext/src/main/kotlin/com/metalbear/mirrord/LogLevel.kt (hunk 1 above)
Comment (eyalb181) [reply intellij-ext/src/main/kotlin/com/metalbear/mirrord/LogLevel.kt:5]: We're probably going to change how our logs are configured anyway (see #390 ), but for now lets keep it as a dropdown, it looks much better and doesn't limit functionality by much.

**Comment by infiniteregrets** [comment]
> hi @jamowei, thanks again for improving the extension!
> 
> Just a quick note on the namespaces,
> 
> They look like this for me: Screen Shot 2022-10-26 at 11 37 05 AM
> 
> even though I have a few more namespaces

this works okay, on a different run.

File: intellij-ext/src/main/kotlin/com/metalbear/mirrord/MirrordDefaultConfig.kt [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import java.nio.file.Paths
+
+data class MirrordDefaultConfig(
+    val ldPreloadPath: String = getSharedLibPath("libmirrord_layer.so"),
+    val dylibPath: String = getSharedLibPath("libmirrord_layer.dylib"),
+    val acceptInvalidCertificates: Boolean = true,
+    val skipProcesses: String = "",
+    val fileOps: Boolean = true,
+    val stealTraffic: Boolean = true,
+    val ephemeralContainers: Boolean = true,
```
Comment (infiniteregrets) [inline intellij-ext/src/main/kotlin/com/metalbear/mirrord/MirrordDefaultConfig.kt:13]: this should be set to false by default

**Review by infiniteregrets — CHANGES_REQUESTED** [review]
small change

**Comment by jamowei** [comment]
> > hi @jamowei, thanks again for improving the extension!
> > Just a quick note on the namespaces,
> > They look like this for me: Screen Shot 2022-10-26 at 11 37 05 AM
> > even though I have a few more namespaces
> 
> this works okay, on a different run.

I did not change the code for receiving the Kubernetes namespaces ;-)

File: intellij-ext/src/main/kotlin/com/metalbear/mirrord/MirrordDefaultConfig.kt (hunk 2 above)
Comment (jamowei) [reply intellij-ext/src/main/kotlin/com/metalbear/mirrord/MirrordDefaultConfig.kt]: Ok, I changed it :)

**Review by infiniteregrets — APPROVED** [review]
Thanks again for the improvements!

**Comment by aviramha** [comment]
This is awesome! Thank you so much.

**Pull request closed without merging**